sotodlib/io/hkdb.py

```diff
@@ -292,12 +299,16 @@ def get_frames_from_file(
         addr = frame['address']
         _, agent, _, feed = addr.split('.')
         start_time, stop_time = 1<<32, 0
+        fields = []
         for block in frame['blocks']:
             ts = np.array(block.times) / spt3g_core.G3Units.s
+            fields.extend(block.keys())
             start_time = min(start_time, ts[0])
             stop_time = max(stop_time, ts[-1])
+
+        fields_hash = hashlib.sha256(','.join(sorted(fields)).encode('ascii')).hexdigest()[:16]
```

Comment:
What's the idea with the `fields_hash` addition to the db?

Reply:
"Hash to identify the combination of fields included in this frame."

To determine what fields are available, we either have to list them all in the database (major expansion) or else read them from the data.  Reading from the data is very slow, since the list of field names can change, so you need to read every frame in the time range.  But agents actually do tend to have the same fields, for long stretches of time.  So the hash allows us to identify a small representative set of frames, where there are potentially a new set of field names.  It keeps the DB lean while also being pretty efficient in minimizing data archive reads to get the full field list.

Reply:
Ahh makes sense, thanks for the explanation!